src/faucet/server_api.rs

```diff
@@ -189,3 +189,137 @@ pub async fn signed_erc20_transfer(
     let signed = sign_with_eth_secret_key(tx.clone(), faucet_info).await?;
     Ok(signed)
 }
+
+#[server(endpoint = "claim_token", input = GetUrl)]
+pub async fn claim_token(
+    faucet_info: FaucetInfo,
+    address: String,
+) -> Result<String, ServerFnError> {
+    use crate::utils::address::{AddressAlloyExt, parse_address};
+    use crate::utils::message::message_transfer;
+    use crate::utils::rpc_context::Provider;
+    use fvm_shared::address::Network;
+    use fvm_shared::address::set_current_network;
+    use leptos_axum::ResponseOptions;
+    use send_wrapper::SendWrapper;
+
+    let response = leptos::prelude::expect_context::<ResponseOptions>();
+
+    SendWrapper::new(async move {
+        match faucet_info {
+            FaucetInfo::MainnetFIL => {
+                response.set_status(http::StatusCode::IM_A_TEAPOT);
+                Err(ServerFnError::ServerError(
+                    "I'm a teapot - mainnet tokens are not available.".to_string(),
+                ))
+            }
+            FaucetInfo::CalibnetFIL => {
+                let network = Network::Testnet;
+                set_current_network(network);
+                let recipient = match parse_address(&address, network) {
```

Comment:
Can you put the parsing logic into a separate method since we are using code twice or just use this:

```
    parse_address(address, network).map_err(|e| {
        log::warn!("Invalid address: {}", e);
        response.set_status(http::StatusCode::BAD_REQUEST);
        ServerFnError::ServerError("Invalid address format".to_string())
    })
```